lib/lib.h

```diff
@@ -169,3 +195,160 @@ unsigned char *mget_address(EC_KEY *ec_key, unsigned int *size_digest) {
 
   return ripemd160_digest;
 }
+
+/* Base58 Encode msg. Takes message, message length and returns the base58 encoded message. 
+   offset is the offset of from where the returned char* has real data 
+*/
+unsigned char *mb58Encode(const unsigned char *msg, int msg_len, int *offset) {
+  // create context for BIGNUMBER operations
+  BN_CTX *ctx = NULL;
+
+  if ((ctx = BN_CTX_new()) == NULL) {
+    printf("Unable to to create BN CTX\n");
+    exit(1);
+  }
+
+  BN_CTX_start(ctx);
+  // Create BIG NUMBER variables
+  BIGNUM *bn58 = NULL;
+  BIGNUM *bn0 = NULL;
+  BIGNUM *bn_msg = NULL;
+  BIGNUM *bn_dv = NULL;
+  BIGNUM *rem = NULL;
+  BIGNUM *temp = NULL;
+
+  // these binary numbers will be converted into Openssl Big Number format
+  const unsigned char bin58 = 58;
+
+  if (!(bn58 = BN_new()) || !(bn0 = BN_new()) || !(bn_msg = BN_new()) || !(bn_dv = BN_new()) || !(rem = BN_new()) || !(temp = BN_new()))
+    printf("Unable to create big numbers for b58 encode");
+
+  // Convert 58, 0 and msg into Big Numbers
+  BN_bin2bn(&bin58, 1, bn58);
+
+  if (!BN_zero(bn0))
+    printf("Unable to set BIGNUM 0");
+  
+  // Convert msg into a bignum to prepare for bignum division
+  BN_bin2bn(msg, msg_len, bn_dv);
+
+  char unsigned bin_rem = 0; // Create remainder variable as a char
+
+  unsigned char *str = malloc(sizeof(char) * BASE58_LEN); // output string
+
+  int i = BASE58_LEN -1;
+
+  str[i--] = '\0'; // create null terminated string
+
+  // while bn_msg as x > 0  divide x by 58
+  while (BN_cmp(bn_dv, bn0) > 0) {
+    if (!BN_copy(temp, bn_dv))
+      error("unable to copy bn_dv to temp");
+
+     // printf("%lu / %lu = ", *temp->d, *bn58->d);
+    if (!BN_div(bn_dv, rem, temp, bn58, ctx))
+      error("Unable to perform BIGNUM division");
+
+     // printf("bn_dv %lu rem: %lu\n", *bn_dv->d, *rem->d);
+
+    BN_bn2bin(rem, &bin_rem); 
+
+    str[i--] = b58[bin_rem]; 
+  }
+
+  // Replace leading zeros in msg hash with the b58 representation of a zero
+  int yes = 0; 
+  do {
+    str[i] = b58[0];
```

Comment:
This is being called everytime adding a '1' (b58[0]) into the begging of the encoded string.  Currently removing this line causes errors.